src/node/eviction.cpp

```diff
@@ -73,15 +73,22 @@ struct CompareNodeNetworkTime {
     };
 };
 
-//! Sort an array by the specified comparator, then erase the last K elements where predicate is true.
-template <typename T, typename Comparator>
-static void EraseLastKElements(
-    std::vector<T>& elements, Comparator comparator, size_t k,
+//! Sort an array of NodeEvictionCandidates by the specified comparator, then erase the last K elements where predicate is true.
+//! Return the last element removed if any.
+static std::optional<NodeEvictionCandidate> EraseLastKElements(
```

Comment:
Returning just the last removed element feels like a very bespoke and perhaps unintuitive interface to me. I know we don't currently need it, but perhaps it's worth generalizing this a bit more and returning all the removed elements instead of just the last one? For the callsite, it's trivial to keep just the last one? Just thinking out loud, perhaps this is difficult to do without an unacceptable overhead in terms of allocations etc.